drivers/dma/dma_stm32.c

```diff
@@ -563,9 +572,12 @@ DMA_STM32_EXPORT_API int dma_stm32_start(const struct device *dev, uint32_t id)
 	}
 
 	dma_stm32_clear_stream_irq(dev, id);
-
 	stm32_dma_enable_stream(dma, id);
 
+	/* When starting the dma, the stream is busy */
+	stream = &config->streams[id];
```

Comment:
Is it possible an irq is fired (since the stream was enabled above) before stream->busy is set?

Reply:
Yes, that could, so inverting setting busy then enabling (when starting and reloading)